lib/chess/board.ex

```diff
@@ -30,13 +30,20 @@ defmodule Chess.Board do
     board["#{file},#{rank}"]
   end
 
-  def move_piece(board, move_params) do
-    [from_file, from_rank] = move_params["from"]
-    [to_file, to_rank] = move_params["to"]
+  def move_piece(board, %{"from" => from, "to" => to}) do
+    [from_file, from_rank] = from
+    [to_file, to_rank] = to
 
     {piece, board} = Map.pop(board, "#{from_file},#{from_rank}")
+    {piece_captured, board} = Map.pop(board, "#{to_file},#{to_rank}")
 
-    Map.put(board, "#{to_file},#{to_rank}", piece)
+    %{
+      from: %{"file" => from_file, "rank" => from_rank},
```

Comment:
This looks like a defined format, what do you think about using a struct here?

Reply:
This structure differs in various parts of the codebase, so it's hard to define a format for it. The main reason for the differences are down to JSON decoding and POST parameters. I might be able to solve the JSON problem by using a different serializer for the database.

Reply:
It would certainly be nice for this to be more consistent!